src/NServiceBus.Core/Transports/Learning/LearningTransportInfrastructure.cs

```diff
@@ -91,16 +91,26 @@ public override TransportSubscriptionInfrastructure ConfigureSubscriptionInfrast
 
         public override string ToTransportAddress(LogicalAddress logicalAddress)
         {
-            var endpoint = logicalAddress.EndpointInstance.Endpoint;
-            PathChecker.ThrowForBadPath(endpoint, "endpoint name");
+            var address = logicalAddress.EndpointInstance.Endpoint;
+            PathChecker.ThrowForBadPath(address, "endpoint name");
 
             var discriminator = logicalAddress.EndpointInstance.Discriminator ?? "";
             PathChecker.ThrowForBadPath(discriminator, "endpoint discriminator");
 
+            if (!string.IsNullOrEmpty(discriminator))
+            {
+                address += "-" + discriminator;
+            }
+
             var qualifier = logicalAddress.Qualifier ?? "";
             PathChecker.ThrowForBadPath(qualifier, "address qualifier");
 
-            return Path.Combine(endpoint, discriminator, qualifier);
+            if (!string.IsNullOrEmpty(qualifier))
```

Comment:
can't this code https://github.com/Particular/NServiceBus/pull/4680/files#diff-050b2ea9fb780fc247eaecef33d051d9R105 also moved into that if statement to simplify it?